flight_planning/v1/flight_planning.yaml

```diff
@@ -475,3 +475,53 @@ paths:
       operationId: DeleteFlightPlan
       description: >-
         This endpoint allows the test director to instruct the USS to remove a flight plan that is no longer needed for testing.
+
+  /user_notifications:
+    get:
+      security:
+        - Authority:
+            - interuss.flight_planning.direct_automated_test
+      parameters:
+        - name: from
+          description: >- 
+            Parameter for filtering the user notifications sent at or after the specified time (inclusive). 
+            The property 'notification_sent_at' in UserNotification is to be used for comparison.
+          schema:
+            type: string
+            format: date-time
+          example: '2024-04-22T16:36:50.52Z'
+          in: query
+          required: true
+        - name: to
+          description: >-
+            Parameter for filtering the user notifications sent till the specified time (inclusive). 
+            This time should be after the time specified in 'to' parameter. 
+            If not provided, then the value defaults to now. 
+            The property 'notification_sent_at' in UserNotification is to be used for comparison.
+          schema:
+            type: string
+            format: date-time
+          example: '2024-04-22T16:41:50.52Z'
+          in: query
+          required: false
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: './user_notification.yaml#/components/schemas/UserNotification'
+          description: Notifications sent to the user were retrieved successfully.
+        '400':
+          description: Query parameter 'from' missing in the request.
+        '401':
+          description: Bearer access token was not provided in Authorization header, token could not be decoded, or token was invalid.
+        '403':
+          description: The access token was decoded successfully but did not include a scope appropriate to this endpoint.
+      summary: User notifications
+      description: >-
+        Returns the list of user notifications, sent by USS, about conflicts in the time interval specified by the query parameters. 
```

Comment:
> But, if the part of the requirement `send within ConflictingOIMaxUserNotificationTime seconds, 95 % of the time` is to be tested later, notification_sent_at would be needed.

No, it would not be needed.  We could verify with higher sensitivity if we had it, but we can still detect non-compliant implementations at some level of sensitivity with only the observation time stamp.  See ["When we're creating these tests" paragraph](https://github.com/interuss/automated_testing_interfaces/pull/19#discussion_r1589491723) and remember that there are manual checkouts capable of verifying these requirements without asking USSs when the notifications were sent -- please consider how that is accomplished.

> Also, it would be helpful if the notification_sent_at and notification_observed_at time points can be indicated in the sequence diagram, to get a clear picture.

Are you saying it is not clear to you where on this diagram these two timestamps would occur?

<img width="628" alt="Screenshot 2024-05-03 at 6 40 39 PM" src="https://github.com/interuss/automated_testing_interfaces/assets/11224086/516e2fbf-7ce5-4acf-897e-03e01bd4bcb6">

Not understanding where those two timestamps would occur on the diagram perhaps explains the overall confusion as labeling them would even further illustrate the problem:

<img width="739" alt="Screenshot 2024-05-03 at 6 32 36 PM" src="https://github.com/interuss/automated_testing_interfaces/assets/11224086/c41dfe0c-4058-4f03-8a0c-c1943631198c">

The problem is that, for the Green user to include the "notification sent at" information in the yellow "Yes" response, that information would have to be transmitted through the black "New conflict" arrow and then again through the black "Pop-up notification" arrow.  Creating a requirement for this transmission (effectively, requiring USSs to display backend-internal information to the user) is not reasonable and not necessary to accomplish the testing objective.

These discussions are consuming a great deal of limited InterUSS resources on what I think are conceptually some of the simplest tests we can perform:

1. Construct conditions that trigger the need for a mandatory notification from the USS
2. Check if that notification was sent

If the high-level concepts are not mapping effectively to lower-level design details (e.g., where the timestamps occur on the above diagram, what information the respondent has available and how it can answer the question we need to answer), it may be useful to adjust strategies to better target all contributors' strengths.  If this PR has not been merged by the Tuesday meeting, let's take that opportunity to see if real-time discussion of this kind of issue is more effective for producing a timely outcome.